src/forms/components/UserForm/index.js

```diff
@@ -0,0 +1,88 @@
+import React, {useState, Fragment} from 'react';
+import {Text, TouchableOpacity} from 'react-native';
+import {Field, reduxForm} from 'redux-form';
+import {useDispatch} from 'react-redux';
+
+import {createFields} from '../../utils';
+import {USER_FORM} from '../../constants';
+
+import actionsCreator from '../../../redux/auth/actions';
+import {required, email, minChar, checkPass} from '../../validate';
+
+import {styles} from './style';
+
+const UserForm = ({navigation, ...props}) => {
+  const [changeForm, setChangeForm] = useState(true);
+  const dispatch = useDispatch();
+
+  const {handleSubmit, reset, submitting} = props;
+
+  const submitLogin = values => {
+    dispatch(actionsCreator.loginToken(navigation, values));
+  };
+
+  const submitRegister = values => {
+    dispatch(actionsCreator.registerToken(navigation, values));
+  };
+
+  const submit = changeForm ? submitLogin : submitRegister;
+  const buttonTxt = changeForm ? 'Login' : 'Register';
+
+  const handleForm = () => {
+    setChangeForm(!changeForm);
+    reset();
```

Comment:
Although this will work here, you should be using the function way for setState if you need the previous state.

```javascript
setChangeForm(prevState => !prevState);
``` 

This is because hooks can be batched and, although extremely rare, it could cause changeForm to be another value than the one you need. Using a function assures the value to be the one you need, in this case, the one before your change.